datasync/chaindatafetcher/kas/repository.go

```diff
@@ -88,3 +97,55 @@ func (r *repository) SetComponent(component interface{}) {
 		r.blockchainApi = c
 	}
 }
+
+func makeEOAListStr(eoaList map[common.Address]struct{}) string {
+	eoaStrList := ""
+	for key := range eoaList {
+		eoaStrList += "\""
+		eoaStrList += strings.ToLower(key.String())
+		eoaStrList += "\""
+		eoaStrList += ","
+	}
+	return eoaStrList[:len(eoaStrList)-1]
+}
+
+func (r *repository) InvalidateCacheEOAList(eoaList map[common.Address]struct{}) {
+	numEOAs := len(eoaList)
+	logger.Trace("the number of EOA list for KAS cache invalidation", "numEOAs", numEOAs)
+	if numEOAs <= 0 || !r.config.CacheUse {
+		return
+	}
+
+	url := r.config.CacheInvalidationURL
+	payloadStr := fmt.Sprintf(`{"type": "stateChange","payload": {"addresses": [%v]}}`, makeEOAListStr(eoaList))
+	payload := strings.NewReader(payloadStr)
+
+	// set up timeout for API call
+	ctx, cancel := context.WithTimeout(context.Background(), apiCtxTimeout)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, "POST", url, payload)
+	if err != nil {
+		logger.Error("Creating a new http request is failed", "err", err, "url", url, "payload", payloadStr)
+		return
+	}
+	req.Header.Add("x-chain-id", r.config.XChainId)
+	req.Header.Add("Authorization", r.config.Authorization)
+	req.Header.Add("Content-Type", "text/plain")
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		logger.Error("Client do method is failed", "err", err, "url", url, "payload", payloadStr, "header", req.Header)
+		return
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			logger.Error("Reading response body is failed", "err", err, "body", res.Body)
+			return
+		}
+		logger.Warn("cache invalidation is failed", "response", string(body))
```

Comment:
We might consider this case as `Error`.

Reply:
Applied. 2e8ba40540932d77628d905c2207db3828f28ffa